internal/util/projutil/exec.go

```diff
@@ -106,10 +119,19 @@ func getGeneralArgs(cmd string, opts GoCmdOptions) ([]string, error) {
 		if goModOn, err := GoModOn(); err != nil {
 			return nil, err
 		} else if goModOn {
-			bargs = append(bargs, "-mod=vendor")
+			// Does vendor exist?
+			info, err := os.Stat("vendor")
+			// Does the first "go" subcommand accept -mod=vendor?
+			_, ok := validVendorCmds[bargs[0]]
+			if err == nil && info.IsDir() && ok {
```

Comment:
One last thing: `os.Stat("vendor")` could fail due to a permissions error, filesystem error, etc.
That means `vendor/` directory exists but with `err != nil` we silently treat it as if `vendor/` does not exist.

Instead we should explicitly return all other errors except `IsNotExist` error by first checking it with [`os.IsNotExist(err)`](https://golang.org/pkg/os/#IsNotExist):

```Go
info, err := os.Stat("vendor")
if err != nil && !os.IsNotExist(err) {
	return nil, err
}

_, ok := validVendorCmds[bargs[0]]
if err == nil && info.IsDir() && ok {
...
}
```